gradio/routes.py

```diff
@@ -592,26 +588,29 @@ def robots_txt():
 ########
 
 
-def safe_join(directory: str, path: str) -> str | None:
-    """Safely path to a base directory to avoid escaping the base directory.
-    Borrowed from: werkzeug.security.safe_join"""
-    _os_alt_seps: List[str] = list(
-        sep for sep in [os.path.sep, os.path.altsep] if sep is not None and sep != "/"
-    )
+def is_relative_to(self, other):
+    # method exist on PurePath class from 3.9
+    try:
+        self.relative_to(other)
+        return True
+    except ValueError:
+        return False
 
-    if path != "":
-        filename = posixpath.normpath(path)
-    else:
-        return directory
 
-    if (
-        any(sep in filename for sep in _os_alt_seps)
-        or os.path.isabs(filename)
-        or filename == ".."
-        or filename.startswith("../")
-    ):
-        return None
-    return posixpath.join(directory, filename)
+def safe_join(directory: str, path: str) -> str:
+    if path == "":
+        raise HTTPException(400)
```

Comment:
This will return an error instead of a 200 when accessing the root path of `directory`

Reply:
Something to add and doable is to check if `path` is requesting a folder and instead of a file